src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -1654,7 +1654,7 @@ trait Namers extends MethodSynthesis {
         if (sym hasAnnotation NativeAttr)
           sym resetFlag DEFERRED
         else {
-          if (!symbolAllowsDeferred && ownerRequiresConcrete) fail(AbstractVar)
+          if (!symbolAllowsDeferred && ownerRequiresConcrete && !sym.isJava) fail(AbstractVar)
```

Comment:
I'm not sure I understand what you mean.

When omitting this change, I get an
```
only classes can have declared but undefined members
        int abstractCommentedMethod);
```
error. However, the error only occurs when a method in an interface has a doc comment. Without a comment it compile just fine. I'll have to revisit that.